src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.IncrementalAnalyzerProcessor.cs

```diff
@@ -155,6 +157,12 @@ private void ResetLogAggregator()
                     _logAggregator = new LogAggregator();
                 }
 
+                private void ReportPendingWorkItemCount()
+                {
+                    var pendingItemCount = _highPriorityProcessor.WorkItemCount + _normalPriorityProcessor.WorkItemCount + _lowPriorityProcessor.WorkItemCount;
+                    _registration.ProgressReporter.UpdatePendingItemCount(pendingItemCount);
```

Comment:
is this threadsafe?

Reply:
are you talking about count? or the event? anyway, count is thread safe. and count can have race you mentioned above. but start/stop shouldn't have race. so at the end, things will get cleaned up.